src/Pandell.PackageRunner/AssemblyExtensions.cs

```diff
@@ -0,0 +1,161 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+using JetBrains.Annotations;
+
+using NuGet;
+
+namespace PackageRunner
+{
+
+    /// <summary>
+    /// Extension methods for <see cref="Assembly"/> instances.
+    /// </summary>
+    /// <remarks>
+    /// Unit tested in <c>Pandell.PackageRunner.Test.AssemblyExtensionsTests.</c>
+    /// </remarks>
+    public static class AssemblyExtensions
+    {
+
+        //--------------------------------------------------
+        /// <summary>
+        /// Adds all embedded assemblies (resources ending with ".dll")
+        /// from the specified assembly to <see cref="AppDomain.AssemblyResolve"/>
+        /// handler (making them available for use without being on disk).
+        /// </summary>
+        public static void EnableResolvingOfEmbeddedAssemblies([NotNull] this Assembly assembly)
+        {
+            if (assembly == null) { throw new ArgumentNullException("assembly"); }
+
+            var embeddedAssemblies = assembly
+                .GetManifestResourceNames()
+                .Where(s => s.EndsWith(".dll"))
+                .Select(s =>
+                {
+                    using (var stream = assembly.GetManifestResourceStream(s))
+                    {
+                        if (stream == null) { throw new InvalidDataException("Embedded resource \"" + s + "\" is null"); }
+                        var assemblyBytes = new byte[stream.Length];
+                        stream.Read(assemblyBytes, 0, assemblyBytes.Length);
+                        return Assembly.Load(assemblyBytes);
+                    }
+                })
+                .ToArray();
+
+            AppDomain.CurrentDomain.AssemblyResolve += (sender, eventArgs) =>
+            {
+                var assemblyToResolve = new AssemblyName(eventArgs.Name);
+                return embeddedAssemblies.FirstOrDefault(
+                    a => AssemblyName.ReferenceMatchesDefinition(assemblyToResolve, a.GetName()));
```

Comment:
It looks like that `ReferenceMatchesDefinition` has a flaw and compares only the names:
From MSDN (https://msdn.microsoft.com/en-us/library/system.reflection.assemblyname.referencematchesdefinition(v=vs.110).aspx#Anchor_1): 
> The comparison depends only on the simple assembly name. It ignores version, culture, and public key token.

And also here they said that they won't fix it to not break existing apps: https://connect.microsoft.com/VisualStudio/feedback/details/752902/assemblyname-referencematchesdefinition-returns-true-even-though-the-parameters-dont-match
Should we revert back to full name string compare?